### ISSUE TITLE
Tests fail on i686

### ISSUE THREAD

**Issue by LucasTavaresA** [philj56/tofi#99]
hello, i maintain this package on void-linux which support a lot of archs, including i686.

the tests added on 9.1 break only on i686, which probably means that tests and maybe tofi does not handle 32bits.
but im not very knowledgeable on c or processor architectures so im not really sure.

the tests log: https://github.com/void-linux/void-packages/actions/runs/4738900963/jobs/8413227989

the full meson log:
```
Build started at 2023-04-30T16:15:40.083768
Main binary: /usr/bin/python3
Build Options: -Db_lto=true -Db_ndebug=true -Db_staticpic=true -Dprefix=/usr -Dbindir=/usr/bin -Ddatadir=/usr/share -Dincludedir=/usr/include -Dinfodir=/usr/share/info -Dlibdir=/usr/lib32 -Dlibexecdir=/usr/libexec -Dlocaledir=/usr/share/locale -Dlocalstatedir=/var -Dmandir=/usr/share/man -Dsbindir=/usr/bin -Dsharedstatedir=/var/lib -Dsysconfdir=/etc -Dauto_features=auto -Dbuildtype=plain -Dwrap_mode=nodownload
Python system: Linux
The Meson build system
Version: 1.1.0
Source dir: /builddir/tofi-0.9.1
Build dir: /builddir/tofi-0.9.1/build
Build type: native build
Project name: tofi
Project version: 0.9.1
-----
Detecting compiler via: cc --version
compiler returned <Popen: returncode: 0 args: ['cc', '--version']>
compiler stdout:
cc (GCC) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


compiler stderr:

Running command: cc -E -dM -
-----
Detecting linker via: cc -Wl,--version -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=.
linker returned <Popen: returncode: 0 args: ['cc', '-Wl,--version', '-Wl,-z,relro', '-Wl,-z,...>
linker stdout:
GNU ld (GNU Binutils) 2.39
Copyright (C) 2022 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.

linker stderr:
collect2 version 12.2.0
/usr/bin/ld -plugin /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/liblto_plugin.so -plugin-opt=/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/lto-wrapper -plugin-opt=-fresolution=/tmp/ccXBFeLQ.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_i386 -dynamic-linker /lib/ld-linux.so.2 -pie /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../Scrt1.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../crti.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/crtbeginS.o -L/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0 -L/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../.. --version -z relro -z now --as-needed -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/crtendS.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../crtn.o

Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -Wl,-z,relro -Wl,-z,now -Wl,--as-needed
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /builddir/tofi-0.9.1/build/meson-private/sanitycheckc.exe
C compiler for the host machine: cc (gcc 12.2.0 "cc (GCC) 12.2.0")
C linker for the host machine: cc ld.bfd 2.39
-----
Detecting compiler via: cc --version
compiler returned <Popen: returncode: 0 args: ['cc', '--version']>
compiler stdout:
cc (GCC) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


compiler stderr:

Running command: cc -E -dM -
-----
Detecting linker via: cc -Wl,--version -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=.
linker returned <Popen: returncode: 0 args: ['cc', '-Wl,--version', '-Wl,-z,relro', '-Wl,-z,...>
linker stdout:
GNU ld (GNU Binutils) 2.39
Copyright (C) 2022 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.

linker stderr:
collect2 version 12.2.0
/usr/bin/ld -plugin /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/liblto_plugin.so -plugin-opt=/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/lto-wrapper -plugin-opt=-fresolution=/tmp/ccmE2Qr1.res -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lgcc_s --build-id --eh-frame-hdr --hash-style=gnu -m elf_i386 -dynamic-linker /lib/ld-linux.so.2 -pie /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../Scrt1.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../crti.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/crtbeginS.o -L/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0 -L/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../.. --version -z relro -z now --as-needed -lgcc --push-state --as-needed -lgcc_s --pop-state -lc -lgcc --push-state --as-needed -lgcc_s --pop-state /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/crtendS.o /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../crtn.o

Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc sanitycheckc.c -o sanitycheckc.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -Wl,-z,relro -Wl,-z,now -Wl,--as-needed
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command:  /builddir/tofi-0.9.1/build/meson-private/sanitycheckc.exe
C compiler for the build machine: cc (gcc 12.2.0 "cc (GCC) 12.2.0")
C linker for the build machine: cc ld.bfd 2.39
Build machine cpu family: x86
Build machine cpu: i686
Host machine cpu family: x86
Host machine cpu: i686
Target machine cpu family: x86
Target machine cpu: i686
Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmpvrkwzm_8
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmpvrkwzm_8/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmpvrkwzm_8/output.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 -Wl,--start-group -lrt -Wl,--end-group -Wl,--allow-shlib-undefined -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 
Library rt found: YES
Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmpw6di5x5z
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmpw6di5x5z/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmpw6di5x5z/output.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 -Wl,--start-group -lm -Wl,--end-group -Wl,--allow-shlib-undefined -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 
Library m found: YES
Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmphgpayfys
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmphgpayfys/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmphgpayfys/output.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 -std=c2x -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 

Code:
 
        #define fts_read meson_disable_define_of_fts_read
        
        #include <limits.h>
        #undef fts_read
        
        #ifdef __cplusplus
        extern "C"
        #endif
        char fts_read (void);
        
        #if defined __stub_fts_read || defined __stub___fts_read
        fail fail fail this function is not going to work
        #endif
        
        int main(void) {
          return fts_read ();
        }
Compiler stdout:
 
Compiler stderr:
 In file included from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/limits.h:26,
                 from /usr/lib/gcc/i686-pc-linux-gnu/12.2/include-fixed/limits.h:203,
                 from /usr/lib/gcc/i686-pc-linux-gnu/12.2/include-fixed/syslimits.h:7,
                 from /usr/lib/gcc/i686-pc-linux-gnu/12.2/include-fixed/limits.h:34,
                 from /builddir/tofi-0.9.1/build/meson-private/tmphgpayfys/testfile.c:4:
/usr/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~

Checking for function "fts_read" : YES 
Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmplwdm_eqq
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmplwdm_eqq/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmplwdm_eqq/output.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 -Wl,--start-group -lfts -Wl,--end-group -Wl,--allow-shlib-undefined -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 

Code:
 int main(void) { return 0; }

Compiler stdout:
 
Compiler stderr:
 /usr/bin/ld: cannot find -lfts: No such file or directory
collect2: error: ld returned 1 exit status

Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmp4ixpj4t6
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmp4ixpj4t6/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmp4ixpj4t6/output.exe -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 -Wl,-z,relro -Wl,-z,now -Wl,--as-needed 

Code:
 
        #include<stdio.h>
        int main(void) {
            printf("%ld\n", (long)(sizeof(void *)));
            return 0;
        }
Compiler stdout:
 
Compiler stderr:
 In file included from /usr/include/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from /builddir/tofi-0.9.1/build/meson-private/tmp4ixpj4t6/testfile.c:2:
/usr/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~

Program stdout:

4

Program stderr:


Running compile:
Working directory:  /builddir/tofi-0.9.1/build/meson-private/tmp405amdfa
Command line:  cc /builddir/tofi-0.9.1/build/meson-private/tmp405amdfa/testfile.c -o /builddir/tofi-0.9.1/build/meson-private/tmp405amdfa/output.obj -c -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=i686 -O2 -pipe -ffile-prefix-map=/builddir/tofi-0.9.1/build=. -D_FILE_OFFSET_BITS=64 -O0 --print-search-dirs 

Code:
 
Compiler stdout:
 install: /usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/
programs: =/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../../i686-pc-linux-gnu/bin/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../../i686-pc-linux-gnu/bin/
libraries: =/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../../i686-pc-linux-gnu/lib/i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../../i686-pc-linux-gnu/lib/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../i686-pc-linux-gnu/12.2.0/:/usr/lib32/gcc/i686-pc-linux-gnu/12.2.0/../../../:/lib/i686-pc-linux-gnu/12.2.0/:/lib/:/usr/lib/i686-pc-linux-gnu/12.2.0/:/usr/lib/

Compiler stderr:
 
Library fts found: NO
Pkg-config binary for 1 is not cached.
Pkg-config binary for 1 specified from cross file, native file, or env var as ['pkg-config']
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Determining dependency 'freetype2' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion freetype2` -> 0
stdout:
24.3.18
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags freetype2` -> 0
stdout:
-I/usr/include/freetype2 -I/usr/include/libpng16
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs freetype2` -> 0
stdout:
-L/usr/lib32 -lfreetype
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs freetype2` -> 0
stdout:
-lfreetype
-----------
Run-time dependency freetype2 found: YES 24.3.18
Pkg-config binary for 1 is cached.
Determining dependency 'harfbuzz' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion harfbuzz` -> 0
stdout:
7.1.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags harfbuzz` -> 0
stdout:
-I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs harfbuzz` -> 0
stdout:
-L/usr/lib32 -lharfbuzz
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs harfbuzz` -> 0
stdout:
-lharfbuzz
-----------
Run-time dependency harfbuzz found: YES 7.1.0
Pkg-config binary for 1 is cached.
Determining dependency 'cairo' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion cairo` -> 0
stdout:
1.16.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags cairo` -> 0
stdout:
-I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs cairo` -> 0
stdout:
-L/usr/lib32 -lcairo
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs cairo` -> 0
stdout:
-lcairo
-----------
Run-time dependency cairo found: YES 1.16.0
Pkg-config binary for 1 is cached.
Determining dependency 'pangocairo' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion pangocairo` -> 0
stdout:
1.50.14
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags pangocairo` -> 0
stdout:
-pthread -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs pangocairo` -> 0
stdout:
-L/usr/lib32 -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs pangocairo` -> 0
stdout:
-lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo
-----------
Run-time dependency pangocairo found: YES 1.50.14
Pkg-config binary for 1 is cached.
Determining dependency 'wayland-client' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion wayland-client` -> 0
stdout:
1.21.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags wayland-client` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-client` -> 0
stdout:
-L/usr/lib32 -lwayland-client
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-client` -> 0
stdout:
-lwayland-client
-----------
Run-time dependency wayland-client found: YES 1.21.0
Pkg-config binary for 0 is not cached.
Pkg-config binary for 0 specified from cross file, native file, or env var as ['pkg-config']
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Determining dependency 'wayland-protocols' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion wayland-protocols` -> 0
stdout:
1.31
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags wayland-protocols` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-protocols` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-protocols` -> 0
Build-time dependency wayland-protocols found: YES 1.31
Pkg-config binary for 0 is cached.
Determining dependency 'wayland-scanner' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion wayland-scanner` -> 0
stdout:
1.21.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags wayland-scanner` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-scanner` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs wayland-scanner` -> 0
Build-time dependency wayland-scanner found: YES 1.21.0
Pkg-config binary for 1 is cached.
Determining dependency 'xkbcommon' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion xkbcommon` -> 0
stdout:
1.5.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags xkbcommon` -> 0
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs xkbcommon` -> 0
stdout:
-L/usr/lib32 -lxkbcommon
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs xkbcommon` -> 0
stdout:
-lxkbcommon
-----------
Run-time dependency xkbcommon found: YES 1.5.0
Pkg-config binary for 1 is cached.
Determining dependency 'glib-2.0' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion glib-2.0` -> 0
stdout:
2.76.1
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags glib-2.0` -> 0
stdout:
-I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs glib-2.0` -> 0
stdout:
-L/usr/lib32 -lglib-2.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs glib-2.0` -> 0
stdout:
-lglib-2.0
-----------
Run-time dependency glib-2.0 found: YES 2.76.1
Pkg-config binary for 1 is cached.
Determining dependency 'gio-unix-2.0' with pkg-config executable '/usr/bin/pkg-config'
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --modversion gio-unix-2.0` -> 0
stdout:
2.76.1
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --cflags gio-unix-2.0` -> 0
stdout:
-pthread -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib32/glib-2.0/include
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_ALLOW_SYSTEM_LIBS]: 1
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs gio-unix-2.0` -> 0
stdout:
-L/usr/lib32 -lgio-2.0 -lgobject-2.0 -lglib-2.0
-----------
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --libs gio-unix-2.0` -> 0
stdout:
-lgio-2.0 -lgobject-2.0 -lglib-2.0
-----------
Run-time dependency gio-unix-2.0 found: YES 2.76.1
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --variable=wayland_scanner wayland-scanner` -> 0
stdout:
/usr/bin/wayland-scanner
-----------
Got pkgconfig variable wayland_scanner : /usr/bin/wayland-scanner
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
env[PKG_CONFIG_FOR_BUILD]: /usr/bin/pkg-config
env[PKG_CONFIG]: pkg-config
env[PKG_CONFIG_PATH]: 
Called `/usr/bin/pkg-config --variable=pkgdatadir wayland-protocols` -> 0
stdout:
//usr/share/wayland-protocols
-----------
Got pkgconfig variable pkgdatadir : //usr/share/wayland-protocols
Adding test "config"
Adding test "utf8"
Program scdoc found: YES (/usr/bin/scdoc)
Program sed found: YES (/usr/bin/sed)
Program sh found: YES (/usr/bin/sh)
Build targets in project: 14

tofi 0.9.1

  User defined options
    auto_features : auto
    bindir        : /usr/bin
    buildtype     : plain
    datadir       : /usr/share
    includedir    : /usr/include
    infodir       : /usr/share/info
    libdir        : /usr/lib32
    libexecdir    : /usr/libexec
    localedir     : /usr/share/locale
    localstatedir : /var
    mandir        : /usr/share/man
    prefix        : /usr
    sbindir       : /usr/bin
    sharedstatedir: /var/lib
    sysconfdir    : /etc
    wrap_mode     : nodownload
    b_lto         : true
    b_ndebug      : true
    b_staticpic   : true

Found ninja-1.11.1 at /usr/bin/ninja
Failed to guess install tag for /usr/share/man/man1/tofi.1
Failed to guess install tag for /usr/share/man/man5/tofi.5
Failed to guess install tag for /etc/xdg/tofi/config
Failed to guess install tag for /usr/share/licenses/tofi/LICENSE
Failed to guess install tag for /usr/share/bash-completion/completions/tofi
Failed to guess install tag for /usr/share/bash-completion/completions/tofi-run
Failed to guess install tag for /usr/share/bash-completion/completions/tofi-drun
Failed to guess install tag for /usr/share/man/man1/tofi-run.1
Failed to guess install tag for /usr/share/man/man1/tofi-drun.1
```

**Comment by philj56** [comment]
Hey, thanks for maintaining it! It's cool that people are interested enough to do that :smile: 

Yep, that's a bug in tofi's parsing - the actually failing tests are:
```
300    ▶ 1/2 - UINT32 -1 FAIL
301    ▶ 1/2 - UINT32 -1 percent without sign FAIL
302    ▶ 1/2 - UINT32 -1 percent with sign FAIL
```
Looks like it's because I was using `strtoul`, which on i686 returns a 32 bit value, but I was treating it as 64 bit — should be fixed in 667075f0920da3c2b353fbce54b6430c195ef031. In any case, those tests are just complaining that an invalid option value is being accepted, so it's not really a problem. I have run tofi on a 32 bit system in the past, and I think it worked fine (but I don't have that system around to test at the moment).

Thanks for reporting this!

**Comment by LucasTavaresA** [comment]
yup tests pass
